<!-- roslaunch seatrac_pkg beacon.launch beacon_type:=X110 working_mode:=SLAVE gps_en:=FALSE tst_comm_en:=TRUE comm_port:=0 -->
<!-- roslaunch seatrac_pkg beacon.launch beacon_type:=X150 working_mode:=POS_TRACK_PING gps_en:=TRUE tst_comm_en:=TRUE comm_port:=0 -->
<!-- roslaunch seatrac_pkg beacon.launch beacon_type:=X150 working_mode:=POS_TRACK_COMM_PING gps_en:=TRUE tst_comm_en:=TRUE comm_port:=0 -->
<!-- roslaunch seatrac_pkg beacon.launch beacon_type:=X150 working_mode:=POS_TRACK_COMM_NAV gps_en:=TRUE tst_comm_en:=TRUE comm_port:=0 -->
<launch>
  <!-- Arguments section (must be passed when running it from console) -->
  <arg name="beacon_type"/>
  <arg name="working_mode"/>
  <arg name="gps_en"/>
  <arg name="tst_comm_en"/>
  <arg name="comm_port"/>
  <node pkg="rosbag" type="record" name="$(anon rosbag_record_beacon)"
       args="record -o $(find seatrac_pkg)/bagfiles/$(arg comm_port)/beacon_$(arg beacon_type) /beacon_pose /beacon_status /beacon_tx_buffer /beacon_rx_buffer /beacon_gps_data /gps"/>
  <!-- Let the console log all the print commands included in the nodes -->
  <env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find seatrac_pkg)/launch/custom_roslogger.conf"/>
  <!-- Beacon node -->
  <node pkg="seatrac_pkg" name="$(anon SeaTracBeacon)" type="Beacon.py" output="screen">
    <param name="beacon_type" type="string" value="$(arg beacon_type)" />
    <param name="working_mode" type="string" value="$(arg working_mode)"/>
    <param name="comm_port" type="string" value="$(arg comm_port)"/>
  </node>
  <!-- GPS Node -->
  <group if="$(arg gps_en)">
    <node pkg="seatrac_pkg" name="GpsTracker" type="GpsTracker.py" output="screen"/>
  </group>
  <!-- Test Communication Node -->
  <group if="$(arg tst_comm_en)">
    <node pkg="seatrac_pkg" name="$(anon TestNode)" type="CustCommTest.py" output="screen">
      <param name="beacon_type" type="string" value="$(arg beacon_type)" />
      <param name="comm_port" type="string" value="$(arg comm_port)"/>
    </node>
  </group>
</launch>
